test/integ/routers/alpha-router/alpha-router.integration.test.ts

```diff
@@ -218,11 +242,22 @@ describe('alpha router integration', () => {
     methodParameters: MethodParameters | undefined,
     tradeType: TradeType,
     checkTokenInAmount?: number,
-    checkTokenOutAmount?: number
+    checkTokenOutAmount?: number,
+    simulate?: boolean
   ) => {
     expect(methodParameters).not.toBeUndefined();
-    const { tokenInBefore, tokenInAfter, tokenOutBefore, tokenOutAfter } =
+    let simulationGasUsed:number|Error = 0
+    if(simulate) {
+      simulationGasUsed = await simulateSwap(tokenIn.chainId,tokenIn.wrapped.address,alice._address,methodParameters!)
+      expect(simulationGasUsed instanceof Error).toBe(false)
+    }
+    const { tokenInBefore, tokenInAfter, tokenOutBefore, tokenOutAfter, gasUsed } =
       await executeSwap(methodParameters!, tokenIn, tokenOut!);
+    
+    if(!(simulationGasUsed instanceof Error)) {
+      console.log(`SIMULATED GAS USED: ${simulationGasUsed},\nACTUAL GAS USED: ${gasUsed}`)
+      expect(Math.abs(gasUsed-simulationGasUsed as number)<0.01*gasUsed).toBe(true);
```

Comment:
I'd add a comment describing the condition you're testing for here. Like `Ensure that simulated gas is within 1% of actual gas`

Also are you using prettier locally?

Reply:
done
and yeah I ran `npx prettier --write src/*`